src/argilla/client/workspaces.py

```diff
@@ -14,67 +13,238 @@
+            id: the ID of the workspace to be managed. Defaults to None.
+
+        Raises:
+            ValueError: if the workspace does not exist in the current Argilla account.
+            ValidationApiError: if the ID provided is not a valid UUID.
+            RuntimeError: if there was an error while retrieving the workspace/s from Argilla.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace("my-workspace")
+            >>> workspace = rg.Workspace(id="...")
+        """
+        if name is None and id is None:
+            raise ValueError("At least one of `name` or `id` must be provided.")
+        if name is not None and id is not None:
+            raise ValueError("Just one of `name` or `id` must be provided.")
+
+        existing_workspace = None
+        if id is not None:
+            try:
+                existing_workspace = workspaces_api_v1.get_workspace(active_client().http_client.httpx, id).parsed
+            except NotFoundApiError as e:
+                raise ValueError(
+                    f"Workspace with id=`{id}` doesn't exist in Argilla, so please"
+                    " make sure that the ID you provided is a valid one. Otherwise,"
+                    " you can create a new one via the `Workspace.create` method."
+                ) from e
+            except ValidationApiError as e:
+                raise ValueError(
+                    "The ID you provided is not a valid UUID, so please make sure"
+                    " that the ID you provided is a valid one."
+                ) from e
+            except BaseClientError as e:
+                raise RuntimeError(f"Error while retrieving workspace with id=`{id}` from Argilla.") from e
+        elif name is not None:
+            try:
+                workspaces = workspaces_api.list_workspaces(active_client().http_client.httpx).parsed
+            except Exception as e:
+                raise RuntimeError("Error while retrieving the list of workspaces from Argilla.") from e
+            for workspace in workspaces:
+                if workspace.name == name:
+                    existing_workspace = workspace
+                    break
+
+            if existing_workspace is None:
+                raise ValueError(
+                    f"Workspace with name=`{name}` doesn't exist in Argilla, so please"
+                    "create it via the `Workspace.create` method as follows:"
+                    f" `Workspace(name=`{name}`)`."
+                )
+        self.__dict__.update(existing_workspace.dict())
+
+    @property
+    def users(self) -> List["WorkspaceUserModel"]:
+        """Returns the list of users linked to the workspace.
+
+        Returns:
+            A list of `WorkspaceUserModel` instances.
+        """
+        if not hasattr(self, "__users") or self.__users is None:
+            self.__users = workspaces_api.list_workspace_users(active_client().http_client.httpx, self.id).parsed
```

Comment:
We should reduce the couple with the active_client logic. I think the httpx client should be part of the workspace on object initialization. Then we can do:
```suggestion
            self.__users = workspaces_api.list_workspace_users(self.__httpx, self.id).parsed
```
otherwise, we can start having problems in multi-thread multi client environments (a process connecting to different argilla server to sync some datasets/workspaces)

Reply:
Sure, then I'll also add the client on top to be reused in the whole class :smile: